executor/distsql.go

```diff
@@ -379,13 +386,34 @@ func (e *IndexLookUpExecutor) Open(ctx context.Context) error {
 		e.feedback.Invalidate()
 		return err
 	}
+	if !e.isImplicitClose {
+		e.originRange = e.ranges
+		e.pageSize = int64(e.ctx.GetSessionVars().IndexLookupSize)
+	}
+	if e.needIndexPaging {
+		if !e.hasAddLimit {
+			e.dagPB.Executors = append(e.dagPB.Executors, e.constructLimitPB(uint64(e.pageSize)))
+			e.hasAddLimit = true
+		} else {
+			e.pageSize = e.pageSize * 2
```

Comment:
When the size reaches the `BatchSize`, the limit is not used anymore. Can we just fall back to the default logic?

Reply:
We need stop IndexScan in time.